### PR TITLE
Added migrations for Self-Serve Migration Integration and API key

### PR DESCRIPTION
**Pull request by naz**
refs https://github.com/TryGhost/Team/issues/2790
refs https://github.com/TryGhost/Ghost/commit/3747df1bc871752d91bc49f0f0f588f6857c4a2e#diff-396038cecd7a381616d00954ae18a655ae2a8af71ea65866bf09d2c7cc1b5235

- This integration will be used to perform self-serve migrations.
- The integration will be limited to these Admin API endpoints: 
```
POST /ghost/api/admin/db
POST /ghost/api/admin/db/media/inline
POST /ghost/api/admin/members/upload
GET  /ghost/api/admin/tags/:id
GET  /ghost/api/admin/tags/slug/:slug
```

### CODE REVIEW

File: ghost/core/core/server/data/schema/fixtures/fixtures.json [hunk 1]
```diff
@@ -701,6 +705,17 @@
                     "type": "core",
                     "api_keys": [{"type": "admin", "role": "Ghost Explore Integration"}]
                 },
+                {
+                    "slug": "self-serve-migration",
+                    "name": "Self-Serve Migration Integration",
+                    "type": "builtin",
```
[the remark this answers is not in the document]
Comment (naz) [reply ghost/core/core/server/data/schema/fixtures/fixtures.json]: We settled on "core" type here and role-based filtering.

File: ghost/core/core/server/data/migrations/versions/5.40/2023-03-21-19-02-add-self-serve-integration-api-key.js [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+        }).first();
+
+        if (existingKey) {
+            logging.warn('Admin API key for "Self-Serve Migration Integration" already exists');
+            return;
+        }
+
+        await knex('api_keys').insert({
+            id: (new ObjectID()).toHexString(),
+            type: 'builtin',
```
Comment (kevinansfield) [inline ghost/core/core/server/data/migrations/versions/5.40/2023-03-21-19-02-add-self-serve-integration-api-key.js:40]: should this be `'core'`?

File: ghost/core/core/server/data/migrations/versions/5.40/2023-03-21-19-02-add-self-serve-integration-api-key.js [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+            created_at: knex.raw('current_timestamp'),
+            created_by: meta.MIGRATION_USER
+        });
+    },
+    async function down(knex) {
+        logging.info('Removing "Self-Serve Migration Integration" API key');
+
+        const integration = await knex('integrations').where({
+            slug: 'self-serve-migration',
+            type: 'builtin',
```
Comment (kevinansfield) [inline ghost/core/core/server/data/migrations/versions/5.40/2023-03-21-19-02-add-self-serve-integration-api-key.js:53]: Has integration type switched to `'core'` now?

File: ghost/core/core/server/data/migrations/versions/5.40/2023-03-21-18-52-add-self-serve-integration.js [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+        }).first();
+
+        if (existingIntegration) {
+            logging.warn('Integration already exists, skipping');
+            return;
+        }
+
+        await knex('integrations').insert({
+            id: (new ObjectID()).toHexString(),
+            type: 'internal',
```
Comment (kevinansfield) [inline ghost/core/core/server/data/migrations/versions/5.40/2023-03-21-18-52-add-self-serve-integration.js:20]: Should this be `'core'`?

File: ghost/core/core/server/data/migrations/versions/5.40/2023-03-21-18-52-add-self-serve-integration.js [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+            slug: 'self-serve-migration',
+            created_at: knex.raw('current_timestamp'),
+            created_by: meta.MIGRATION_USER
+        });
+    },
+    async function down(knex) {
+        logging.info('Deleting "Self-Serve Migration Integration"');
+
+        await knex('integrations').where({
+            type: 'internal',
```
Comment (kevinansfield) [inline ghost/core/core/server/data/migrations/versions/5.40/2023-03-21-18-52-add-self-serve-integration.js:32]: As above 🙂

File: ghost/core/core/server/data/migrations/versions/5.40/2023-03-21-19-02-add-self-serve-integration-api-key.js (hunk 2 above)
Comment (naz) [reply ghost/core/core/server/data/migrations/versions/5.40/2023-03-21-19-02-add-self-serve-integration-api-key.js:40]: yes, 100% on all of these 😅

**Pull request merged**